site/source/docs/porting/Rust.rst

```diff
@@ -0,0 +1,51 @@
+.. Rust:
+
+=================
+Rust + Emscripten
+=================
+
+The
+`Rust programming language <https://www.rust-lang.org/>`_
+has direct support for compiling to WebAssembly.
+In addition to that, Rust can be used with Emscripten, which allows Rust to
+benefit from Emscripten's API support. For example, using Emscripten with Rust
+can allow you to port a Rust codebase using :ref:`OpenGL <OpenGL-support>` or
+:ref:`Pthreads <Pthreads-support>`, as Emscripten has library support code for
+those APIs which handles the large differences between them and the closest Web
+APIs.
+
+============
+Installation
+============
+
+The Rust toolchain will look for ``emcc`` and the other Emscripten components
+in the path. You should therefore first
+:ref:`download, install, and add Emscripten to your path <sdk-download-and-install>`.
+
+It is crucial to install the proper versions of the tools. Specifically, Rust
+uses a particular LLVM version, and Emscripten must use the same one, so that
+object files and so forth are all compatible.
+
+As of June 2021, stable Rust is ``1.52.1``, which uses LLVM 12, which was
+branched off of the ``llvmorg-12-init`` tag. That LLVM change rolled into
+Emscripten in
+`56b877afe7d1b651d6b9a2ba5d5df074876172ff <https://chromium.googlesource.com/emscripten-releases/+/56b877afe7d1b651d6b9a2ba5d5df074876172ff>`_,
```

Comment:
Does this wording mean this is the earliest version where LLVM 12 was rolled into Emscripten? If so, perhaps we can advise on the latest one that is compatible so that users get newer features and bugfixes too?

Reply:
We don't in general know which llvm commit is the latest compatible one, since any commit after the branch point could introduce some incompatibility in principle.

Reply:
Right, what I meant is whether we can get latest hash before the next LLVM roll. Or do we roll LLVM continuously in Emscripten?

Reply:
LLVM is rolled continuously. So there might be release after this one with some Binaryen change before the next release with an LLVM change, but the difference would probably not be meaningful.

Reply:
Ah okay, makes sense.